geth/account/accounts_test.go

```diff
@@ -103,9 +107,381 @@ func TestVerifyAccountPasswordWithAccountBeforeEIP55(t *testing.T) {
+		s.NoError(err)
+	})
+
+	s.T().Run("ReSelect_fail", func(t *testing.T) {
+		s.reinitMock()
+		s.nodeManager.EXPECT().WhisperService().Return(nil, testErrWhisper).AnyTimes()
+		err = s.accManager.ReSelectAccount()
+		s.Error(err)
+	})
+
+	s.accManager.selectedAccount = nil
+	s.reinitMock()
+	s.nodeManager.EXPECT().AccountKeyStore().Return(s.keyStore, nil).AnyTimes()
+	s.nodeManager.EXPECT().WhisperService().Return(nil, testErrWhisper).AnyTimes()
+
+	s.T().Run("Selected_fail", func(t *testing.T) {
+		_, err := s.accManager.SelectedAccount()
+		s.Equal(ErrNoAccountSelected, err)
+	})
+
+	s.T().Run("ReSelect_success_noAccount", func(t *testing.T) {
+		err = s.accManager.ReSelectAccount()
+		s.NoError(err)
+	})
+}
+
+func (s *ManagerTestSuite) TestLogout() {
+	s.reinitMock()
+
+	s.nodeManager.EXPECT().WhisperService().Return(s.shh, nil)
+	err := s.accManager.Logout()
+	s.NoError(err)
+
+	s.nodeManager.EXPECT().WhisperService().Return(nil, testErrWhisper)
+	err = s.accManager.Logout()
+	s.Error(err)
+}
+
+func (s *ManagerTestSuite) TestAccounts() {
+	s.reinitMock()
+
+	// Select the test account
+	s.nodeManager.EXPECT().AccountKeyStore().Return(s.keyStore, nil).AnyTimes()
+	s.nodeManager.EXPECT().WhisperService().Return(s.shh, nil).AnyTimes()
+	err := s.accManager.SelectAccount(s.address, s.password)
+	s.NoError(err)
+
+	// Success
+	s.nodeManager.EXPECT().AccountManager().Return(s.gethAccManager, nil)
+	accs, err := s.accManager.Accounts()
+	s.NoError(err)
+	s.NotNil(accs)
+
+	// Can't get an account manager
+	s.nodeManager.EXPECT().AccountManager().Return(nil, errors.New("Can't return an account manager"))
+	_, err = s.accManager.Accounts()
+	s.Error(err)
+
+	// Selected account is nil but doesn't fail
+	s.accManager.selectedAccount = nil
+	s.nodeManager.EXPECT().AccountManager().Return(s.gethAccManager, nil)
+	accs, err = s.accManager.Accounts()
+	s.NoError(err)
+	s.NotNil(accs)
+}
+
+func (s *ManagerTestSuite) TestAddressToDecryptedAccount() {
+	s.reinitMock()
+
+	testCases := []struct {
+		name                  string
+		accountKeyStoreReturn []interface{}
+		address               string
+		password              string
+		fail                  bool
+	}{
+		{
+			"success",
+			[]interface{}{s.keyStore, nil},
+			s.address,
+			s.password,
+			false,
+		},
+		{
+			"fail_keyStore",
+			[]interface{}{nil, testErrKeyStore},
+			s.address,
+			s.password,
+			true,
+		},
+		{
+			"fail_wrongAddress",
+			[]interface{}{s.keyStore, nil},
+			"wrong-address",
+			s.password,
+			true,
+		},
+		{
+			"fail_wrongPassword",
+			[]interface{}{s.keyStore, nil},
+			s.address,
+			"wrong-password",
+			true,
+		},
+	}
+
+	for _, testCase := range testCases {
+		s.T().Run(testCase.name, func(t *testing.T) {
+			s.reinitMock()
+			s.nodeManager.EXPECT().AccountKeyStore().Return(testCase.accountKeyStoreReturn...).AnyTimes()
+			acc, keyStore, err := s.accManager.AddressToDecryptedAccount(testCase.address, testCase.password)
```

Comment:
We should be able to verify `acc` value.

Reply:
This looks like the best way to do it but please confirm (there's a typo - `keyStore` should be `key`):
```go
s.Equal(acc.Address, key.Address)
s.keyStore.Find(acc)
```

Reply:
`s.Equal(acc.Address, key.Address)` check should be ok.

Regarding, `s.keyStore.Find(acc)`, per `AddressToDecryptedAccount` comment, it requires the key file for a given address to be prior to node start. So, I think it's not necessary as this call should fail if this check is invalid anyway.